jax/experimental/export/_shape_poly.py

```diff
@@ -586,19 +596,20 @@ def get_vars(self) -> set[str]:
       acc.update(mon.get_vars())
     return acc
 
-  @classmethod
-  def _linear_combination(
-      cls,
+  @staticmethod
```

Comment:
Some staticmethods in this PR are effectively specialized constructors, but this one should probably just be a standalone function. Wdyt?

Reply:
Then only reason this is in `_DimExpr` is to make it clear that it applies to expressions.